examples/conversation.v1.js

```diff
@@ -13,24 +13,51 @@ const conversation = new watson.ConversationV1({
 });
 
 /**
- * Payload for the Watson Conversation Service
- * <workspace-id> and user input text required.
+ * Conversation service is stateless.
+ * The application need to maintain the conversation state.
+ * A context object is returned in the messgage response and the app should pass it in the next message request
  */
-const payload = {
-  workspace_id: process.env.WORKSPACE_ID || '<workspace_id>',
-  input: {
-    text: 'I am happy to talk to you today.'
-  }
+let currentContext = undefined;
+
+/**
+ * Calls the conversation message api. 
+ * returns a promise
+ */
+const message = function(text) {
+  const payload = {
+    workspace_id: process.env.WORKSPACE_ID || '<workspace_id>',
+    input: {
+      text: text
+    },
+    // pass the context from previous response
+    context: currentContext
+  };
+  return new Promise((resolve, reject) => conversation.message(payload, function(err, data) {
+    if (err) {
+      reject(err);
+    } else {
+      // remember the conversation state
+      currentContext = data.context;
+      resolve(data);
+    }
+  }));
 };
 
-conversation.message(payload, function(err, data) {
-  if (err) {
-    // APPLICATION-SPECIFIC CODE TO PROCESS THE ERROR
-    // FROM CONVERSATION SERVICE
-    console.error(JSON.stringify(err, null, 2));
-  } else {
-    // APPLICATION-SPECIFIC CODE TO PROCESS THE DATA
-    // FROM CONVERSATION SERVICE
-    console.log(JSON.stringify(data, null, 2));
-  }
+// This example makes two successive calls to conversation service.
+message('first message').then(response1 => {
+  // APPLICATION-SPECIFIC CODE TO PROCESS THE DATA
+  // FROM CONVERSATION SERVICE
+  console.log(JSON.stringify(response1, null, 2));
+
+  // invoke a second call to conversation
+  message('second message')
+    .then(response2 => {
```

Comment:
I think you should return `message('second message')` here and move the `.then()` and `.catch()` to the outside. That way the `.catch()` handle errors from either message.